fdbserver/TenantBalancer.actor.cpp

```
@@ -918,7 +933,24 @@ ACTOR Future<Void> finishSourceMovement(TenantBalancer* self, FinishSourceMoveme
 
 	try {
 		// TODO: check that the DR is ready to switch
-		// TODO: check if maxLagSeconds is exceeded
+
+		// Check if maxLagSeconds is exceeded
+		state std::vector<TenantMovementInfo> filteredMovements = wait(getFilteredMovements(
+		    self, Optional<std::string>(req.sourceTenant), Optional<std::string>(), Optional<MovementLocation>()));
+		if (filteredMovements.size() != 1) {
+			throw movement_not_found();
+		}
+		TenantMovementInfo targetMovementInfo = filteredMovements[0];
+		if (targetMovementInfo.mutationLag > req.maxLagSeconds) {
+			TraceEvent(SevDebug, "TenantBalancerLagSecondsCheckFailed")
+			    .detail("MaxLagSeconds", req.maxLagSeconds)
+			    .detail("CurrentLagSeconds", targetMovementInfo.mutationLag);
+			return Void();
```

We have to also send a reply to the sender, probably by throwing an appropriate error rather than returning here.